express-get/server.ts

```diff
@@ -0,0 +1,66 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+// connection is already established
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/pagila',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.get('/api/countries', async (req, res, next) => {
+  try {
+    const sql = `select  count(*) as "cities",
+                 "countryId", "countries"."name"
+                 from "countries" join "cities"
+                 using ("countryId")
+                 group by "countries"."countryId";`; // group by to something unique where name causes an error
+
+    const result = await db.query(sql);
+    const countries = result.rows;
+    // or res.send(countries)
+    res.json(countries); // send() it sends back res in json
+    // but we want to explicitly showing we are sending in json
+  } catch (error) {
+    next(error);
+  }
+});
+
+app.get('/api/cities/:cityId', async (req, res, next) => {
+  try {
+    const { cityId } = req.params;
+    if (!Number.isInteger(+cityId) || cityId === undefined) {
+      throw new ClientError(400, 'bad request');
```

Comment:
It is helpful to tell the client why the request is bad, so they have a hope of figuring out what they did wrong.